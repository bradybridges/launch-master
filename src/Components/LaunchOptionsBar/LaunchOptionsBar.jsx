import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { FormControl, InputLabel, Select, MenuItem } from "@material-ui/core";
import ToggleButton from "@material-ui/lab/ToggleButton";
import ToggleButtonGroup from "@material-ui/lab/ToggleButtonGroup";

export default function LaunchOptionsBar({
  setTimeFrame,
  setNumResults,
  numResults,
  timeframe,
}) {
  const [active, setActive] = React.useState("upcoming");

  const useStyles = makeStyles((theme) => ({
    buttonContainer: {
      display: "flex",
      alignItems: "center",
      flexDirection: "column",
      padding: `${theme.spacing(10)}px 0 0 0`,
    },
    formControl: {
      margin: theme.spacing(1),
      minWidth: 120,
    },
    selectEmpty: {
      marginTop: theme.spacing(2),
    },
  }));

  const handleNumResults = (e) => {
    setNumResults(e.target.value);
  };

  const handleTimeFrame = (newTimeFrame) => {
    if (newTimeFrame === timeframe) return;

    if (newTimeFrame === "upcoming") {
      setActive("upcoming");
      setTimeFrame(false);
    } else {
      setActive("past");
      setTimeFrame(true);
    }
  };

  const classes = useStyles();
  return (
    <div
      className={classes.buttonContainer}
      aria-label="select upcoming launches or past launches"
    >
      <ToggleButtonGroup
        value={active}
        exclusive
        aria-label="past or upcoming launch selection"
      >
        <ToggleButton
          value="upcoming"
          aria-label="upcoming launches"
          onClick={() => handleTimeFrame("upcoming")}
          selected={!timeframe}
        >
          Upcoming
        </ToggleButton>
        <ToggleButton
          value="past"
          aria-label="past launches"
          onClick={() => handleTimeFrame("past")}
          selected={timeframe}
        >
          Past
        </ToggleButton>
      </ToggleButtonGroup>
      {active === "upcoming" ? (
        <FormControl variant="filled" className={classes.formControl}>
          <InputLabel>Results</InputLabel>
          <Select
            labelId="filled-label"
            id="select-filled"
            value={numResults}
            onChange={handleNumResults}
          >
            <MenuItem value={10}>10</MenuItem>
            <MenuItem value={25}>25</MenuItem>
            <MenuItem value={50}>50</MenuItem>
          </Select>
        </FormControl>
      ) : null}
    </div>
  );
}
