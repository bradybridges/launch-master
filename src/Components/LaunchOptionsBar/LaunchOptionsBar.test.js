import React from "react";
import { shallow } from "enzyme";
import { FormControl, InputLabel, Select, MenuItem } from "@material-ui/core";
import ToggleButton from "@material-ui/lab/ToggleButton";
import ToggleButtonGroup from "@material-ui/lab/ToggleButtonGroup";
import LaunchOptionsBar from "./LaunchOptionsBar";

describe("LaunchOptionsBar", () => {
  const mockSetTimeFrame = jest.fn();
  const mockSetNumResults = jest.fn();
  const wrapper = shallow(
    <LaunchOptionsBar
      setTimeFrame={mockSetTimeFrame}
      setNumResults={mockSetNumResults}
      numResults={10}
    />
  );

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Toggle button group should have expected props", () => {
    const buttonGroup = wrapper.find(ToggleButtonGroup).first().props();
    expect(buttonGroup.value).toEqual("upcoming");
    expect(buttonGroup["aria-label"]).toEqual(
      "past or upcoming launch selection"
    );
  });

  it("Upcoming launches button should have expected properties and call setTimeFrame when clicked", () => {
    const upcomingButton = wrapper.find(ToggleButton).at(0).props();
    expect(upcomingButton.value).toEqual("upcoming");
    expect(upcomingButton["aria-label"]).toEqual("upcoming launches");
    expect(upcomingButton.children).toEqual("Upcoming");
    upcomingButton.onClick();
    expect(mockSetTimeFrame).toHaveBeenCalledWith(false);
  });

  it("Past launches button should have expected props and call setTimeFrame when clicked", () => {
    const pastButton = wrapper.find(ToggleButton).at(1).props();
    expect(pastButton.value).toEqual("past");
    expect(pastButton.children).toEqual("Past");
    expect(pastButton["aria-label"]).toEqual("past launches");
    pastButton.onClick();
    expect(mockSetTimeFrame).toHaveBeenCalledWith(true);
    const upcomingButton = wrapper.find(ToggleButton).at(0).props();
    upcomingButton.onClick();
  });

  it("FormControl should have expected properties", () => {
    const formControl = wrapper.find(FormControl).first().props();
    expect(formControl.variant).toEqual("filled");
  });

  it("InputLabel should have a value of Results", () => {
    const inputLabel = wrapper.find(InputLabel).first().props();
    expect(inputLabel.children).toEqual("Results");
  });

  it("Select element should have the correct properties", () => {
    const select = wrapper.find(Select).first().props();
    expect(select.id).toEqual("select-filled");
    expect(select.value).toEqual(10);
  });

  it("Select element should call handleNumResults when the user selects a new value", () => {
    const select = wrapper.find(Select).first().props();
    select.onChange({ target: { value: 25 } });
    expect(mockSetNumResults).toHaveBeenCalledWith(25);
  });

  it("There should be three menu items with values 10, 25, and 50", () => {
    const menuItems = wrapper.find(MenuItem);
    expect(menuItems.length).toEqual(3);
    const itemOne = menuItems.at(0).props();
    const itemTwo = menuItems.at(1).props();
    const itemThree = menuItems.at(2).props();
    expect(itemOne).toEqual({ value: 10, children: "10" });
    expect(itemTwo).toEqual({ value: 25, children: "25" });
    expect(itemThree).toEqual({ value: 50, children: "50" });
  });
});
