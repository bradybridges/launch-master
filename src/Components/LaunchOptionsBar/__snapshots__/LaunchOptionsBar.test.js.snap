// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchOptionsBar should match snapshot 1`] = `
<div
  aria-label="select upcoming launches or past launches"
  className="makeStyles-buttonContainer-1"
>
  <WithStyles(ForwardRef(ToggleButton))
    aria-label="past or upcoming launch selection"
    exclusive={true}
    value="upcoming"
  >
    <WithStyles(ForwardRef(ToggleButton))
      aria-label="upcoming launches"
      onClick={[Function]}
      selected={true}
      value="upcoming"
    >
      Upcoming
    </WithStyles(ForwardRef(ToggleButton))>
    <WithStyles(ForwardRef(ToggleButton))
      aria-label="past launches"
      onClick={[Function]}
      value="past"
    >
      Past
    </WithStyles(ForwardRef(ToggleButton))>
  </WithStyles(ForwardRef(ToggleButton))>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-formControl-2"
    variant="filled"
  >
    <WithStyles(ForwardRef(InputLabel))>
      Results
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(Select))
      id="select-filled"
      labelId="filled-label"
      onChange={[Function]}
      value={10}
    >
      <WithStyles(ForwardRef(MenuItem))
        value={10}
      >
        10
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value={25}
      >
        25
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value={50}
      >
        50
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
  </WithStyles(ForwardRef(FormControl))>
</div>
`;
